zstd_stream_test.go

```diff
@@ -139,6 +139,8 @@ func BenchmarkStreamCompression(b *testing.B) {
 		if err != nil {
 			b.Fatalf("Failed writing to compress object: %s", err)
 		}
+		// Prevent from unbound buffer growth.
+		intermediate.Reset()
```

Comment:
The `Reset` must be in the loop, since otherwise:
- The performance of the code inside loop will depend on the value of `b.N`. This contradicts the benchmarking rules and makes the results non-determenistic. See https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go .
- The amount of required memory for each benchmark run will be proportional to `b.N`. This may lead to out of memory error on a system with low amount of memory.

The [bytes.Buffer.Reset](https://golang.org/src/bytes/buffer.go?s=3634:3658#L88) skews the benchmark results by a  few nanoseconds. This is a noise comparing to 20 microseconds spent on the data compression at the line 138.